src/bonescript.js

```diff
@@ -7,7 +7,9 @@ _bonescript.on.connect = function () {};
 _bonescript.on.connecting = function () {};
 _bonescript.on.disconnect = function () {};
 _bonescript.on.connect_failed = function () {};
-_bonescript.on.error = function () {};
+_bonescript.on.error = function (err) {
+    throw (new Error(err))
```

Comment:
This runs asynchronously on load when loaded by a browser? How do we catch this error?

Reply:
I am not very sure how we will be able to catch this error from the browser , however if the current implementation is used, then a 401 Unauthorized response could be sent for unauthorized sessions from browser.